src/net/sourceforge/kolmafia/request/AutoSellRequest.java

```diff
@@ -108,15 +108,17 @@ public ArrayList<TransferItemRequest> generateSubInstances() {
 
     List<AdventureResult> all = new ArrayList<>();
     List<AdventureResult> allButOne = new ArrayList<>();
-    Set<AdventureResult> others = new HashSet<AdventureResult>();
-
-    for (int index = 0; index < this.attachments.length; ++index) {
-      AdventureResult item = this.attachments[index];
+    Set<AdventureResult> others = new HashSet<>();
 
+    for (AdventureResult item : this.attachments) {
       if (item == null) {
         continue;
       }
 
+      if (item.getCount() == 0L) {
```

Comment:
Given AdventureResult and the long version, this doesn't seem aesthetically correct.  I think using getCount and not getLongCount is correct but I would test against 0 and not 0L

Reply:
good spot! yes, had this as `getLongCount` to start with, adjusted to long, then didn't re-adjust.